jobs/dtd-metrics/src/services/tenant-distribution-metric.service.ts

```diff
@@ -0,0 +1,84 @@
+import { AgreementState, ReadModelClient } from '@interop-be-reports/commons'
+import { TenantDistributionMetric } from '../models/metrics.model.js'
+import uniq from 'lodash/uniq.js'
+import { z } from 'zod'
+
+export async function getTenantDistributionMetric(readModel: ReadModelClient): Promise<TenantDistributionMetric> {
+  const onBoardedTenantIdsQueryPromise = readModel.tenants
+    .find({ 'data.selfcareId': { $exists: true } }, { projection: { _id: 0, 'data.id': 1 } })
+    .map(({ data }) => z.string().parse(data.id))
+    .toArray()
+
+  const activeAgreementsQueryPromise = readModel.agreements
+    .find(
+      { 'data.state': { $in: ['Active', 'Suspended'] satisfies Array<AgreementState> } },
+      {
+        projection: {
+          _id: 0,
+          'data.producerId': 1,
+          'data.consumerId': 1,
+        },
+      }
+    )
+    .map(({ data }) =>
+      z
+        .object({
+          producerId: z.string(),
+          consumerId: z.string(),
+        })
+        .parse(data)
+    )
+    .toArray()
+
+  const [onBoardedTenantIds, activeAgreements] = await Promise.all([
+    onBoardedTenantIdsQueryPromise,
+    activeAgreementsQueryPromise,
+  ])
+
+  const agreementsConsumersIds = uniq(activeAgreements.map((agreement) => agreement.consumerId))
+  const agreementsProducersIds = uniq(activeAgreements.map((agreement) => agreement.producerId))
```

Comment:
may be worth to use [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#performance) in this case?

Reply:
Yes, why not 😁